hermes/probe/journal/journal.go

```diff
@@ -0,0 +1,364 @@
+	}
+	j.checksum = fmt.Sprintf("%x", h.Sum(nil))
+	return nil
+}
+
+// validateChecksum validates that the checksum matches the checksum
+// in the file name.
+// The checksum in the filename is being validated against the file contents
+// This will return false if the file contents have been updated
+// without the file name being changed accordingly.
+// Returns:
+//	- err: returns err if the checksum does not match or the checksum or filename is blank, else nil.
+func (j *journalFile) validateChecksum() *probe.ProbeError {
+	if j.checksum == "" {
+		return probe.NewProbeError(metrics.ProbeFailed, fmt.Errorf("validateChecksum(): journal checksum cannot be validated as checksum, %q, is blank", j.checksum))
+	}
+	if j.name == "" {
+		return probe.NewProbeError(metrics.ProbeFailed, fmt.Errorf("validateChecksum(): journal checksum cannot be validated as name, %q, is blank", j.name))
+	}
+
+	if j.checksum != j.name[len(journalFilePrefix):] {
+		return probe.NewProbeError(metrics.ProbeFailed, fmt.Errorf("validateChecksum(): checksum %q does not match checksum in name %q", j.checksum, j.name[len(journalFilePrefix):]))
+	}
+
+	return nil
+}
+
+// reader returns an io reader that in turn returns the file contents.
+// Returns:
+// - reader: returns an io reader that in turn returns the file contents.
+func (j *journalFile) reader() io.Reader {
+	return bytes.NewReader(j.contents)
+}
+
+// unmarshalContents unmarshals the contents of the journal file and returns
+// the completed proto struct.
+// Returns:
+// 		- journal: proto struct containing the unmarshalled contents of the journal
+//		- err: error if once occurs during the unmarshalling process.
+func (j *journalFile) unmarshalContents() (*pb.StateJournal, *probe.ProbeError) {
+	journal := &pb.StateJournal{}
+	if err := proto.Unmarshal(j.contents, journal); err != nil {
+		return nil, probe.NewProbeError(metrics.FileReadFailure, fmt.Errorf("verifyNilFile(): unable to unmarshal nil file proto: %w", err))
+	}
+	return journal, nil
+}
+
+// CheckJournal checks the existence and validity of the Journal file on the target
+// storage system. It also reads the intent from the Journal file to identify if
+// Hermes failed to complete the last operation, and completes it if necessary.
+// Arguments:
+//	- ctx: context so this probe can be cancelled if needed.
+//	- target: target run information.
+//	- client: initialised storage client for this target system.
+//	- logger: logger associated with the probe calling this function.
+// Returns:
+//	- err: error exit statuses are well defined in metrics.go
+func CheckJournal(ctx context.Context, target *probe.Target, client stiface.Client, logger *logger.Logger) *probe.ProbeError {
+	file, err := findJournal(ctx, target, client, logger)
+	if err != nil {
+		return err
+	}
+
+	if file.contents, err = readJournal(ctx, file.name, target, client, logger); err != nil {
```

Comment:
The `file` variable is constructed with two function calls (findJournal, readJournal).
This makes it a bit hard to follow what happens.

To simplify it, I would change this to

```
fileName, err := findJournal(...)
if err != nil {
  ...
}

contents, err := readJournal(...)
if err != nil {
  ...
}

j := &journalFile{
  name: fileName,
  contents: contents
}
```

Reply:
I've amended the code to follow this structure.